File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utils/Angle.java
package org.firstinspires.ftc.teamcode.utils;

public class Angle {
    public static final float pi = (float) Math.PI;
    public static final float twoPi = (float) Math.PI * 2;

    public static float wrapPlusMinusPI(final float theta) {
        final float wrapped = theta % twoPi;

        if (wrapped > pi) {
            return wrapped - twoPi;
        } else if (wrapped < -pi) {
            return wrapped + twoPi;
        }

        return wrapped;

    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/snav/util/Angle.kt
package org.firstinspires.ftc.teamcode.snav.util

import kotlin.math.abs

class Angle(angle: Double) {
    var angle: Double = angle
        set(value) {
            field = normalizeAngle(value)
        }

    // Normalize the angle [-180, -180]
    fun normalizeAngle(x: Double): Double {
        return ((x % 360 + 360) % 360) - 180
    }

    fun getDistance(other: Angle): Double {
        return abs(other.angle - angle)
    }

    fun getMinimumRotation(other: Angle): Double {
        val difference: Double = other.angle - angle
        return normalizeAngle(difference)
    }

    override fun toString(): String {
        return "Angle: $angle"
    }
}